flatbuffers: 1.8.0 -> 1.10.0

r-ryantm opened the pull request:
Semi-automatic update generated by https://github.com/ryantm/nixpkgs-update tools. This update was made based on information from https://repology.org/metapackage/flatbuffers/versions.



meta.description for flatbuffers is: '"Memory Efficient Serialization Library."'.

[Release on GitHub](https://github.com/google/flatbuffers/releases/tag/v1.10.0)


[Compare changes on GitHub](https://github.com/google/flatbuffers/compare/v1.8.0...v1.10.0)

<details>
<summary>
Checks done (click to expand)
</summary>

- built on NixOS
- /nix/store/2y14hsb7z8qf1p7rwh92wgqcbjpji021-flatbuffers-1.10.0/bin/flatc passed the binary check.
- 1 of 1 passed binary check by having a zero exit code.
- 0 of 1 passed binary check by having the new version present in output.
- found 1.10.0 with grep in /nix/store/2y14hsb7z8qf1p7rwh92wgqcbjpji021-flatbuffers-1.10.0
- directory tree listing: https://gist.github.com/564af87c4b11bab6768480fff2842c75
- du listing: https://gist.github.com/dac87fb9f9ad12518d695e6c55ae498e

</details>
<details>
<summary>
Instructions to test this update (click to expand)
</summary>

One-time optional setup to skip building using Cachix:
```
nix-shell -p cachix --run 'cachix use r-ryantm'
```

Test this update by entering a nix shell, seeing what is inside the
result, and if applicable, running some binaries:
```
nix-shell --pure -I nixpkgs=https://github.com/r-ryantm/nixpkgs/archive/df47e0b82a19491804693155d4005acb90978cb1.tar.gz -p flatbuffers
ls -la /nix/store/2y14hsb7z8qf1p7rwh92wgqcbjpji021-flatbuffers-1.10.0
```

</details>
<br/>


cc @teh for testing.

r-ryantm committed:
flatbuffers: 1.8.0 -> 1.10.0

Semi-automatic update generated by
https://github.com/ryantm/nixpkgs-update tools. This update was made
based on information from
https://repology.org/metapackage/flatbuffers/versions

GrahamcOfBorg commented:
<!--REQUEST_ID=c548d7d3-3829-4fe4-b689-508a6bb5b1ac-->
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.48521&attempt_id=2e0d544e-1ca5-4d9a-9c5e-a9ac4f814123)

Attempted: flatbuffers

<details><summary>Partial log (click to expand)</summary><p>

```
building '/nix/store/42zfrxciis3b77asp43fpi0w94z743qp-flatbuffers-1.10.0.drv'...
unpacking sources
unpacking source archive /nix/store/qp8yjp2ls2ybjvrxy6mncr0nhji2fni8-source
source root is source
patching sources
configuring
fixing cmake files...
mkdir: cannot create directory 'build': File exists
builder for '/nix/store/42zfrxciis3b77asp43fpi0w94z743qp-flatbuffers-1.10.0.drv' failed with exit code 1
error: build of '/nix/store/42zfrxciis3b77asp43fpi0w94z743qp-flatbuffers-1.10.0.drv' failed
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=c548d7d3-3829-4fe4-b689-508a6bb5b1ac-->
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.48521&attempt_id=22ba1395-9464-4781-902e-2417f08485f3)

Attempted: flatbuffers

<details><summary>Partial log (click to expand)</summary><p>

```
-- Installing: /nix/store/d9cmwgqm379cyayfjph5kybdk9r91qww-flatbuffers-1.10.0/lib/cmake/flatbuffers/FlatcTargets.cmake
-- Installing: /nix/store/d9cmwgqm379cyayfjph5kybdk9r91qww-flatbuffers-1.10.0/lib/cmake/flatbuffers/FlatcTargets-release.cmake
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/d9cmwgqm379cyayfjph5kybdk9r91qww-flatbuffers-1.10.0
shrinking /nix/store/d9cmwgqm379cyayfjph5kybdk9r91qww-flatbuffers-1.10.0/bin/flatc
strip is /nix/store/p9akxn2sfy4wkhqdqa3li97pc6jaz3r1-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/d9cmwgqm379cyayfjph5kybdk9r91qww-flatbuffers-1.10.0/lib  /nix/store/d9cmwgqm379cyayfjph5kybdk9r91qww-flatbuffers-1.10.0/bin
patching script interpreter paths in /nix/store/d9cmwgqm379cyayfjph5kybdk9r91qww-flatbuffers-1.10.0
checking for references to /build in /nix/store/d9cmwgqm379cyayfjph5kybdk9r91qww-flatbuffers-1.10.0...
/nix/store/d9cmwgqm379cyayfjph5kybdk9r91qww-flatbuffers-1.10.0
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=c548d7d3-3829-4fe4-b689-508a6bb5b1ac-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.48521&attempt_id=f20f6238-3aed-45f5-9d37-b65a728c7cfb)

Attempted: flatbuffers

<details><summary>Partial log (click to expand)</summary><p>

```
-- Installing: /nix/store/2y14hsb7z8qf1p7rwh92wgqcbjpji021-flatbuffers-1.10.0/lib/cmake/flatbuffers/FlatcTargets.cmake
-- Installing: /nix/store/2y14hsb7z8qf1p7rwh92wgqcbjpji021-flatbuffers-1.10.0/lib/cmake/flatbuffers/FlatcTargets-release.cmake
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/2y14hsb7z8qf1p7rwh92wgqcbjpji021-flatbuffers-1.10.0
shrinking /nix/store/2y14hsb7z8qf1p7rwh92wgqcbjpji021-flatbuffers-1.10.0/bin/flatc
strip is /nix/store/vcc4svb8gy29g4pam2zja6llkbcwsyiq-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/2y14hsb7z8qf1p7rwh92wgqcbjpji021-flatbuffers-1.10.0/lib  /nix/store/2y14hsb7z8qf1p7rwh92wgqcbjpji021-flatbuffers-1.10.0/bin
patching script interpreter paths in /nix/store/2y14hsb7z8qf1p7rwh92wgqcbjpji021-flatbuffers-1.10.0
checking for references to /build in /nix/store/2y14hsb7z8qf1p7rwh92wgqcbjpji021-flatbuffers-1.10.0...
/nix/store/2y14hsb7z8qf1p7rwh92wgqcbjpji021-flatbuffers-1.10.0
```
</p></details>

c0bw3b commented:
Related: #38731 #48066

Still failing on Darwin though..

badi commented:
The darwin failure is likely due to the case-insensitive filesystem clashing with the bazel `BUILD` file.

I was able to get it to build by adding

```
preConfigure = lib.optional stdenv.isDarwin ''
  rm BUILD
'';
```

c0bw3b committed:
flatbuffers: fix build on Darwin and enable checks

Bazel BUILD file is conflicting with build directory on case-insensitive FS
+ checkPhase now runs "make test"

GrahamcOfBorg commented:
<!--REQUEST_ID=fe4cc41a-34c3-4795-a384-ebe239a56955-->
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.48521&attempt_id=a53624c1-4246-4b46-8c20-61a7414ac430)

Attempted: flatbuffers

<details><summary>Partial log (click to expand)</summary><p>

```
-- Installing: /nix/store/0wfkd6fpqv140bqd8wk9nfcf6pd7g74l-flatbuffers-1.10.0/lib/cmake/flatbuffers/FlatcTargets.cmake
-- Installing: /nix/store/0wfkd6fpqv140bqd8wk9nfcf6pd7g74l-flatbuffers-1.10.0/lib/cmake/flatbuffers/FlatcTargets-release.cmake
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/0wfkd6fpqv140bqd8wk9nfcf6pd7g74l-flatbuffers-1.10.0
shrinking /nix/store/0wfkd6fpqv140bqd8wk9nfcf6pd7g74l-flatbuffers-1.10.0/bin/flatc
strip is /nix/store/p9akxn2sfy4wkhqdqa3li97pc6jaz3r1-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/0wfkd6fpqv140bqd8wk9nfcf6pd7g74l-flatbuffers-1.10.0/lib  /nix/store/0wfkd6fpqv140bqd8wk9nfcf6pd7g74l-flatbuffers-1.10.0/bin
patching script interpreter paths in /nix/store/0wfkd6fpqv140bqd8wk9nfcf6pd7g74l-flatbuffers-1.10.0
checking for references to /build in /nix/store/0wfkd6fpqv140bqd8wk9nfcf6pd7g74l-flatbuffers-1.10.0...
/nix/store/0wfkd6fpqv140bqd8wk9nfcf6pd7g74l-flatbuffers-1.10.0
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=fe4cc41a-34c3-4795-a384-ebe239a56955-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.48521&attempt_id=4c927994-41aa-4caa-b963-d4ccb993cce2)

Attempted: flatbuffers

<details><summary>Partial log (click to expand)</summary><p>

```
-- Installing: /nix/store/f84zzdmh2fa9zn2i65dgj1n48i6a48vx-flatbuffers-1.10.0/lib/cmake/flatbuffers/FlatcTargets.cmake
-- Installing: /nix/store/f84zzdmh2fa9zn2i65dgj1n48i6a48vx-flatbuffers-1.10.0/lib/cmake/flatbuffers/FlatcTargets-release.cmake
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/f84zzdmh2fa9zn2i65dgj1n48i6a48vx-flatbuffers-1.10.0
shrinking /nix/store/f84zzdmh2fa9zn2i65dgj1n48i6a48vx-flatbuffers-1.10.0/bin/flatc
strip is /nix/store/vcc4svb8gy29g4pam2zja6llkbcwsyiq-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/f84zzdmh2fa9zn2i65dgj1n48i6a48vx-flatbuffers-1.10.0/lib  /nix/store/f84zzdmh2fa9zn2i65dgj1n48i6a48vx-flatbuffers-1.10.0/bin
patching script interpreter paths in /nix/store/f84zzdmh2fa9zn2i65dgj1n48i6a48vx-flatbuffers-1.10.0
checking for references to /build in /nix/store/f84zzdmh2fa9zn2i65dgj1n48i6a48vx-flatbuffers-1.10.0...
/nix/store/f84zzdmh2fa9zn2i65dgj1n48i6a48vx-flatbuffers-1.10.0
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=fe4cc41a-34c3-4795-a384-ebe239a56955-->
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.48521&attempt_id=77e63b9c-7aec-4dc8-8ab7-bba82faacda1)

Attempted: flatbuffers

<details><summary>Partial log (click to expand)</summary><p>

```
-- Installing: /nix/store/h8y14ddm1q8kmhzjayciz902piqazzrr-flatbuffers-1.10.0/lib/cmake/flatbuffers/FlatbuffersTargets.cmake
-- Installing: /nix/store/h8y14ddm1q8kmhzjayciz902piqazzrr-flatbuffers-1.10.0/lib/cmake/flatbuffers/FlatbuffersTargets-release.cmake
-- Installing: /nix/store/h8y14ddm1q8kmhzjayciz902piqazzrr-flatbuffers-1.10.0/bin/flatc
-- Installing: /nix/store/h8y14ddm1q8kmhzjayciz902piqazzrr-flatbuffers-1.10.0/lib/cmake/flatbuffers/FlatcTargets.cmake
-- Installing: /nix/store/h8y14ddm1q8kmhzjayciz902piqazzrr-flatbuffers-1.10.0/lib/cmake/flatbuffers/FlatcTargets-release.cmake
post-installation fixup
strip is /nix/store/g5r4apl0za012ffs6ladinwa5w0m1l3k-cctools-binutils-darwin/bin/strip
stripping (with command strip and flags -S) in /nix/store/h8y14ddm1q8kmhzjayciz902piqazzrr-flatbuffers-1.10.0/lib  /nix/store/h8y14ddm1q8kmhzjayciz902piqazzrr-flatbuffers-1.10.0/bin
patching script interpreter paths in /nix/store/h8y14ddm1q8kmhzjayciz902piqazzrr-flatbuffers-1.10.0
/nix/store/h8y14ddm1q8kmhzjayciz902piqazzrr-flatbuffers-1.10.0
```
</p></details>
